tests/ui_tests/app/page/host_list/page.py

```diff
@@ -178,6 +178,38 @@ def _check_host_state(page: HostListPage, row: WebElement):
         host_row = self.table.get_row(row_num)
         wait_until_step_succeeds(_check_host_state, timeout=10, period=0.5, page=self, row=host_row)
 
+    @allure.step('Click on maintenance mode button in row {row_num}')
+    def click_on_maintenance_mode_btn(self, row_num: int):
+        """Click maintenance mode in row"""
+
+        row = self.table.get_row(row_num)
+        self.find_child(row, HostListLocators.HostTable.HostRow.maintenance_mode_btn).click()
+
+    @allure.step('Assert maintenance mode state in row {row_num}')
+    def assert_maintenance_mode_state(self, row_num: int, allow_mm_state: bool = True):
+        """Assert maintenance mode state in row"""
+
+        def _check_mm_state(page: HostListPage, row: WebElement):
+            button_state = page.find_child(row, HostListLocators.HostTable.HostRow.maintenance_mode_btn).get_attribute(
+                "class"
+            )
+            tooltips_info = [
+                t.get_property("innerHTML") for t in page.find_elements(HostListLocators.HostTable.tooltip_text)
+            ]
+            if allow_mm_state:
+                assert "mat-primary" in button_state, "Button should be gray"
+                assert (
+                    "Turn maintenance mode ON" in tooltips_info
+                ), "There should be tooltip that user could turn on maintenance mode"
+            else:
+                assert "mat-on" in button_state, "Button should be red"
+                assert (
+                    "Turn maintenance mode OFF" in tooltips_info
+                ), "There should be tooltip that user could turn off maintenance mode"
```

Comment:
What does `allow_mm_state` mean?

Reply:
renamed as is_mm_state_on

Reply:
Then what's about the "disabled" state? If it shouldn't be checked by this method, then please note that in docstring.
Otherwise you may want to use constants from func tests' conftest file (maybe move it to an upper level), so we have similar API.

Reply:
added for disable state